src/formatting.rs

```diff
@@ -683,6 +701,45 @@ fn parse_crate(
     Err(ErrorKind::ParseError)
 }
 
+struct SilentOnIgnoredFilesEmitter {
+    ignore_path_set: IgnorePathSet,
+    source_map: Rc<SourceMap>,
+    emitter: EmitterWriter,
+    can_reset: bool,
+    has_non_ignorable_parser_errors: bool,
+    parser_error_resetter: Rc<RefCell<bool>>,
```

Comment:
Seems that [Handler](https://docs.rs/rustc-ap-rustc_errors/583.0.0/rustc_errors/struct.Handler.html) requires a [Boxed Emitter with a static lifetime](https://docs.rs/rustc-ap-rustc_errors/583.0.0/rustc_errors/struct.Handler.html#method.with_emitter_and_flags) and I could not see a way around the conflicting lifetime requirements to be able to use a mutable reference:
https://github.com/rust-lang/rust/blob/52b3d1f2eb339a2c414b7a18bdf5331308a69192/src/librustc_errors/lib.rs#L305

Reply:
Ideally we can somehow and and use a getter method on `SilentOnIgnoredFilesEmitter` instead of the mutable reference. Unfortunately doing so seems quite hard given that `Handler` will erase the type to `Emitter` 🤷‍♂

Reply:
Agreed! The other problem is that the Handler doesn't publicly expose access to its emitter either